toolkit/tools/internal/file/file.go

```diff
@@ -89,6 +89,77 @@ func Copy(src, dst string) (err error) {
 	return NewFileCopyBuilder(src, dst).Run()
 }
 
+// CopyDir copies src directory to dst, creating the dst directory if needed.
+// dst is assumed to be a directory and not a file.
+func CopyDir(src, dst string, newDirPermissions, mergedDirPermissions, childFilePermissions *fs.FileMode) (err error) {
+	isDstExist, err := PathExists(dst)
+	if err != nil {
+		return err
+	}
+	if isDstExist {
+		isDstDir, err := IsDir(dst)
+		if err != nil {
+			return err
+		}
+		if isDstDir {
+			logger.Log.Infof("Destination (%s) already exists and is a directory", dst)
+			if mergedDirPermissions != nil {
+				if err := os.Chmod(dst, *mergedDirPermissions); err != nil {
+					return fmt.Errorf("error setting file permissions: %w", err)
+				}
+			}
+		}
+	}
+
+	if !isDstExist {
+		logger.Log.Infof("Creating destination directory on chroot (%s)", dst)
+		// Create dst dir
+		if newDirPermissions != nil {
+			err = os.MkdirAll(dst, *newDirPermissions)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = os.MkdirAll(dst, 0755)
```

Comment:
nit: Move `0755` into a constant, instead of repeating the value everywhere.

Reply:
Added constant.